tools/consumedir-cleaner/consumedir-cleaner.sh

```diff
@@ -0,0 +1,179 @@
+    login
+    printf "%${#len_resultset}s" " "; printf "           .."
+    curl_call $1
+
+  elif [ "$curl_result" == "Bad Gateway" ] || [ "$curl_result" == '404 page not found' ]; then
+    echo "FATAL  Connection to server failed"
+    exit -1
+  fi
+}
+
+
+function login() {
+  curl_call "curl -s -X POST -d '{\"account\": \"$ds_collective/$ds_user\", \"password\": \"$ds_password\"}' ${ds_url}/api/v1/open/auth/login"
+
+  curl_status=$(echo $curl_result | jq -r ".success")
+
+  if [ "$curl_status" == "true" ]; then
+    ds_token=$(echo $curl_result | jq -r ".token")
+    echo "Login successfull ( Token: $ds_token )"
+
+  else
+    echo "FATAL  Login not succesfull"
+    exit 1
+
+  fi
+}
+
+############# END
+
+echo "Settings:"
+if [ "$DS_CC_REMOVE" == "true" ]; then
+  echo "    ### !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! ###"
+  echo "  - DELETE files?    YES"
+  echo "     when already existing in Docspell. This cannot be undone!"
+  echo "    ### !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! ###"
+else
+  echo "  - DELETE files?    no"
+  echo "     moving already uploaded files to archive"
+fi
+echo
+if [ "$DS_CC_UPLOAD_MISSING" == true ]; then
+  echo "  - UPLOAD files?    YES"
+  echo "     files not existing in Docspell will be uploaded and will be re-checked in the next run."
+else
+  echo "  - UPLOAD files?    no"
+  echo "     files not existing in Docspell will NOT be uploaded and stay where they are."
+fi
+echo && echo
+echo "Press 'ctrl+c' to cancel"
+for ((i=9;i>=0;i--)); do
+  printf "\r waiting $i seconds "
+  sleep 1s
+done
+echo && echo
+
+# login, get token
+login
+
+echo "Scanning folder for collective '$ds_collective' ($ds_consumedir_path/$ds_collective)"
+echo && echo
+
+while read -r line
+do
+  tmp_filepath=$line
+
+  if [ "$tmp_filepath" == "" ]; then
+    echo "no files found" && echo
+    exit 0 #no results
+  elif [ ! -f "$tmp_filepath" ]; then
+    echo "FATAL  no access to file: $tmp_filepath"
+    exit 3
+  fi
+
+  echo "Checking '$tmp_filepath'"
+  printf "%${#len_resultset}s" " "; printf "           "
+
+  # check for checksum
+  tmp_checksum=$(sha256sum "$tmp_filepath" | awk '{print $1}')
+
+  curl_call "curl -s -X GET '$ds_url/api/v1/sec/checkfile/$tmp_checksum'"
+  curl_status=$(echo $curl_result | jq -r ".exists")
+
+  if [ $curl_code -ne 0 ]; then
+    # error
+    echo "ERROR  $curl_result // $curl_status"
+
+  # file exists in Docspell
+  elif [ "$curl_status" == "true" ]; then
+    item_name=$(echo $curl_result | jq -r ".items[0].name")
+    item_id=$(echo $curl_result | jq -r ".items[0].id")
+    echo "File already exists: '$item_name (ID: $item_id)'"
+
+    printf "%${#len_resultset}s" " "; printf "           "
+    if [ "$DS_CC_REMOVE" == "true" ]; then
+      echo "... removing file"
+      rm "$tmp_filepath"
+    else
+      created=$(echo $curl_result | jq -r ".items[0].created")
+      cur_dir="$ds_archive_path/$(date -d @$(echo "(1603892183000+500)/1000" | bc) +%Y-%m
```

Comment:
Isn't this echo thing doing a constant calculation; did you mean to write `$created` maybe?

Reply:
Thanks for showing me the `bc` command btw!

Reply:
of course this wasn't meant to be, thanks!

Yeah, `bc` is quite powerful